src/derivative_operators/BC_operators.jl

```diff
@@ -139,14 +145,150 @@ function LinearAlgebra.Array(Q::BoundaryPaddedVector)
+# A multidim convolution would be better. - this is unused in DifEqOperators.jl but used as a template for the following functions
+function slicemul(A, u::AbstractArray, dim::Integer)
+    s = size(u)
+    out = similar(u)
+    if dim == 1
+        for j in 1:size[3]
+            for i in 1:size[2]
+                @views out[:,i,j] = A*u[:,i,j]
+            end
+        end
+    elseif dim == 2
+        for j in 1:size[3]
+            for i in 1:size[1]
+                @views out[i,:,j] = A*u[i,:,j]
+            end
+        end
+    elseif dim == 3
+        for j in 1:size[2]
+            for i in 1:size[1]
+                @views out[i,j,:] = A*u[i,j,:]
+            end
+        end
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return out
+end
+
+
+# The BC is applied stripwise and the boundary Arrays built from the l/r of the BoundaryPaddedVectors
+@inline function slicemul(A::SingleLayerBC, u::AbstractArray{T, 2}, dim::Integer) where T
+    s = size(u)
+    if dim == 1
+        lower = zeros(T, s[2])
+        upper = deepcopy(lower)
+        for i in 1:s[2]
+            tmp = A*u[:,i]
+            lower[i] = tmp.l
+            upper[i] = tmp.r
+        end
+    elseif dim == 2
+        lower = zeros(T, s[1])
+        upper = deepcopy(lower)
+        for i in 1:s[1]
+            tmp = A*u[i,:]
+            lower[i] = tmp.l
+            upper[i] = tmp.r
+        end
+    elseif dim == 3
+        throw("The 3 dimensional Method should be being called, not this one. Check dispatch.")
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return lower, upper
+end
+
+@inline function slicemul(A::SingleLayerBC, u::AbstractArray{T, 3}, dim::Integer) where T
+    s = Array(size(u))
+    if dim == 1
+        lower = zeros(T, s[2], s[3])
+        upper = deepcopy(lower)
+        for j in 1:s[3]
+            for i in 1:s[2]
+                tmp = A*u[:,i,j]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    elseif dim == 2
+        lower = zeros(T, s[1], s[3])
+        upper = deepcopy(lower)
+        for j in 1:s[3]
+            for i in 1:s[1]
+                tmp = A*u[i,:,j]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    elseif dim == 3
+        lower = zeros(T, s[1], s[2])
+        upper = deepcopy(lower)
+        for j in 1:s[2]
+            for i in 1:s[1]
+                tmp = A*u[i,j,:]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return lower, upper
+end
+
+struct MultiDimensionalSingleLayerBC{T, N} <: MultiDimensionalBC{T, N}
+    BC::Vector{SingleLayerBC{T}} # I think this has to be an array of non concrete BCs to allow different BCs on different dims
```

Comment:
each point along the boundary can have a different BC, so maybe this needs to be an Vector{Array{SingleLayerBC{T},N-1}}. For example, 2D, [1] is a vector of BCs for the left/right BCs, and [2] is the vector of BCs for the top bottom. 3D, [1] is a matrix of left right BCs at each point in [y,z], [2] is matrix of top bottom bcs in [x,z], and [3] is a matrix of front back BCs in [x,y]. The higher dimensional extension of this interpretation is then clear.

Reply:
That makes sense, I'll try to move towards this more flexible implementation next time i sit down with the code